cmd/osquery-perf/agent.go

```diff
@@ -654,9 +659,42 @@ func (a *agent) mdm() []map[string]string {
 	}
 }
 
+var munkiIssues = func() []string {
+	// generate a list of random munki issues (messages)
+	issues := make([]string, 1000)
+	for i := range issues {
+		// message size: between 60 and 200, with spaces between each 10-char word so
+		// that it can still make a bit of sense for UI tests.
+		numParts := rand.Intn(15) + 6 // number between 0-14, add 6 to get between 6-20
+		var sb strings.Builder
+		for j := 0; j < numParts; j++ {
+			if j > 0 {
+				sb.WriteString(" ")
+			}
+			sb.WriteString(randomString(10))
+		}
+		issues[i] = sb.String()
+	}
+	return issues
+}()
+
 func (a *agent) munkiInfo() []map[string]string {
+	var errors, warnings []string
+
+	if rand.Float64() <= a.munkiIssueProb {
+		for i := 0; i < a.munkiIssueCount; i++ {
+			if rand.Intn(2) == 1 {
+				errors = append(errors, munkiIssues[rand.Intn(len(munkiIssues))])
+			} else {
+				warnings = append(warnings, munkiIssues[rand.Intn(len(munkiIssues))])
+			}
+		}
+	}
+
+	errList := strings.Join(errors, ";")
+	warnList := strings.Join(warnings, ";")
 	return []map[string]string{
-		{"version": "1.2.3"},
+		{"version": "1.2.3", "errors": errList, "warnings": warnList},
 	}
```

Comment:
(random thought) At some point we should probably see about abstracting these away into something more reusable so adding new ones gets easier.

Reply:
Do you mean the error/warning messages in particular or you're referring to the generation of the data in general?

Reply:
Generating of data that has a randomness factor.

Reply:
Created the ticket: https://github.com/fleetdm/fleet/issues/7375